workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-etc-cron/src/main/java/com/liferay/marketplace/MarketplaceCommandLineRunner.java

```diff
@@ -0,0 +1,208 @@
+import org.springframework.web.reactive.function.client.WebClient;
+
+/**
+ * @author Wellington Barbosa
+ */
+@Component
+public class MarketplaceCommandLineRunner implements CommandLineRunner {
+
+	public void run(String... args) throws Exception {
+		_processExpiredTrials();
+		_processOnHoldTrials();
+	}
+
+	private void _addTrial(Order order) throws Exception {
+		WebClient.create(
+			_marketplaceSpringBootUrl
+		).post(
+		).uri(
+			"/trial/provisioning"
+		).header(
+			HttpHeaders.AUTHORIZATION,
+			_liferayOAuth2AccessTokenManager.getAuthorization(
+				"liferay-marketplace-etc-spring-boot-oauth-application-" +
+					"headless-server")
+		).accept(
+			MediaType.APPLICATION_JSON
+		).contentType(
+			MediaType.APPLICATION_JSON
+		).bodyValue(
+			new JSONObject(
+			).put(
+				"classPK", order.getId()
+			).put(
+				"modelDTOOrder",
+				new JSONObject(
+				).put(
+					"accountId",
+					order.getAccountId(
+					).toString()
+				)
+			).toString()
+		).retrieve(
+		).bodyToMono(
+			String.class
+		).block();
+	}
+
+	private void _deleteTrial(long orderId) throws Exception {
+		WebClient.create(
+			_marketplaceSpringBootUrl
+		).delete(
+		).uri(
+			"/trial/" + orderId
+		).header(
+			HttpHeaders.AUTHORIZATION,
+			_liferayOAuth2AccessTokenManager.getAuthorization(
+				"liferay-marketplace-etc-spring-boot-oauth-application-" +
+					"headless-server")
+		).accept(
+			MediaType.APPLICATION_JSON
+		).retrieve(
+		).bodyToMono(
+			String.class
+		).block();
+	}
+
+	private JSONObject _getAvailabilityJSONObject() throws Exception {
+		return new JSONObject(
+			WebClient.create(
+				_marketplaceSpringBootUrl
+			).get(
+			).uri(
+				"/trial/availability"
+			).header(
+				HttpHeaders.AUTHORIZATION,
+				_liferayOAuth2AccessTokenManager.getAuthorization(
+					"liferay-marketplace-etc-spring-boot-oauth-application-" +
+						"headless-server")
+			).accept(
+				MediaType.APPLICATION_JSON
+			).retrieve(
+			).bodyToMono(
+				String.class
+			).block());
+	}
+
+	private OrderResource _getOrderResource() throws Exception {
+		URL liferayDXPURL = new URL(
+			_lxcDXPServerProtocol + "://" + _lxcDXPMainDomain);
+
+		return OrderResource.builder(
+		).endpoint(
+			liferayDXPURL
+		).header(
+			HttpHeaders.AUTHORIZATION,
+			_liferayOAuth2AccessTokenManager.getAuthorization(
+				"liferay-marketplace-etc-spring-boot-oauth-application-" +
+					"headless-server")
+		).build();
+	}
+
+	private void _processExpiredTrials() throws Exception {
+		Page<Order> ordersPage = _getOrderResource().getOrdersPage(
+			"", "orderStatus/any(x:(x eq " + _ORDER_STATUS_COMPLETED + "))",
+			Pagination.of(-1, -1), "");
+
+		for (Order order : ordersPage.getItems()) {
+			Date expirationDate = Date.from(
+				ZonedDateTime.parse(
+					order.getCustomFields(
+					).get(
+						"trial-end-date"
+					).toString()
+				).toInstant());
+
+			boolean expired = new Date(
+			).after(
+				expirationDate
+			);
+
+			if (expired) {
```

Comment:
Inline

```suggestion
			if (new Date(
			).after(
				expirationDate
			)) {
```

@nNilton I think we tried to inline this first, but SF was throwing some weird error, this inline is necessary here?